src/map/map.ts

```diff
@@ -386,9 +392,40 @@ export class PropertiesMap {
         assert(markerData !== undefined);
 
         const indexes = this._indexer.from_environment(markerData.current);
-        this.onselect(indexes, this._active);
+        this.activate(this._active);
     }
 
+    /**
+     * Removes a marker from the map.
+     *
+     * The parameter deleteFromGUIDs pertains to removing the guid from the guid
+     * list. This is *not* the same as the `force` parameter in
+     * `structure._removeWidget`. One may want to remove a marker when switching
+     * from 2D --> 3D but retain the information.
```

Comment:
I am not entirely happy with this: the function seems to do both hiding of HTML markers when switching from 2D to 3D; and marker removal. Could these two actions be separated easily?